static/js/components/widgets/ResourcePickerDialog.tsx

```diff
@@ -5,42 +5,82 @@ import Dialog from "../Dialog"
 import {
   RESOURCE_TYPE_DOCUMENT,
   RESOURCE_TYPE_IMAGE,
-  RESOURCE_TYPE_VIDEO
+  RESOURCE_TYPE_VIDEO,
+  CONTENT_TYPE_RESOURCE,
+  CONTENT_TYPE_PAGE
 } from "../../constants"
 import ResourcePickerListing from "./ResourcePickerListing"
 import { useDebouncedState } from "../../hooks/state"
 import {
   CKEResourceNodeType,
-  ResourceDialogState,
-  RESOURCE_EMBED
+  ResourceDialogMode,
+  RESOURCE_EMBED,
+  RESOURCE_LINK
 } from "../../lib/ckeditor/plugins/constants"
 import { WebsiteContent } from "../../types/websites"
 
 interface Props {
-  state: ResourceDialogState
+  mode: ResourceDialogMode
+  isOpen: boolean
   closeDialog: () => void
   insertEmbed: (id: string, title: string, variant: CKEResourceNodeType) => void
-  attach: string
 }
 
-const RESOURCE_PICKER_TABS = [
+interface ResourceTabSettings {
+  title: string
+  id: string
+  contentType: "resource" | "page"
+  resourcetype: string | null
+  canEmbed: boolean
+}
+const RESOURCE_PICKER_TABS: ResourceTabSettings[] = [
```

Comment:
Basically all the functionality for this feature was already present, we just needed a way for the tab to differentiate whether it should make API calls with `type=resource` or `type=page`. This array controls those settings for the tabs.